pkg/asset/installconfig/ssh.go

```diff
@@ -25,6 +25,14 @@ func (a *sshPublicKey) Dependencies() []asset.Asset {
 
 // Generate generates the SSH public key asset.
 func (a *sshPublicKey) Generate(map[asset.Asset]*asset.State) (state *asset.State, err error) {
+	if value, ok := os.LookupEnv("OPENSHIFT_INSTALL_SSH_PUB_KEY"); ok {
```

Comment:
I'd consider to also ask for the SSH key if not provided, or are there use-cases where one wouldn't want one set?

Reply:
> I'd consider to also ask for the SSH key if not provided, or are there use-cases where one wouldn't want one set?

I'm fine allowing folks to not set one (e.g. if they trust us to set up a working cluster for them so they won't have to poke around ;).  So I think we want to allow empty `OPENSHIFT_INSTALL_SSH_PUB_KEY` to mean "I don't want to set this, don't ask me".  Alternatively, we could have a `--batch` flag or some such to mean "anything I haven't passed to you (via environment variables or otherwise) is not coming, so don't ask."

Reply:
Setting `OPENSHIFT_INSTALL_SSH_PUB_KEY` to an empty string effectively boots without any keys. There is an empty key, but that's okay.

Reply:
r

> Alternatively, we could have a `--batch` flag or some such to mean "anything I haven't passed to you (via environment variables or otherwise) is not coming, so don't ask."

That sounds useful. If someone goes through it the first time interactively they might forget or not be aware of this setting and later have to restart when they find themselves unable to login.

Reply:
> If someone goes through it the first time interactively they might forget or not be aware of this setting and later have to restart when they find themselves unable to login.

The long-term plan is to have one of the operators (the machine-config operator?) managing authorized SSH keys over the life of the cluster.  So you should (eventually) be able to recover even if you launch a cluster without this.